File: modules/core/src/main/java/kr/entree/enderwand/command/CommandContext.java
package kr.entree.enderwand.command;

import kr.entree.enderwand.command.sender.Sender;
import lombok.Getter;
import lombok.Setter;

/**
 * Created by JunHyung Lim on 2020-02-04
 */
public class CommandContext<S extends Sender> {
    @Getter
    private final S sender;
    @Getter
    private final StringReader reader;
    @Getter
    @Setter
    private CommandArguments arguments = CommandArguments.EMPTY;

    public CommandContext(S sender, StringReader reader) {
        this.sender = sender;
        this.reader = reader;
    }
}

File: modules/core-kotlin/src/main/kotlin/enderwand/command/CommandContext.kt
package enderwand.command

import enderwand.command.sender.Sender

/**
 * Created by JunHyung Lim on 2020-01-08
 */
open class CommandContext<S : Sender, T>(
    val sender: S,
    val args: T
) {
    val metadata: MutableMap<String, Any> by lazy { mutableMapOf<String, Any>() }
}